README.md

```diff
@@ -114,6 +114,57 @@ It will then restart the child process.
 
 This way you can add a crontab entry to force a restart so you don't have to monitor the process all the time.
 
+hooks
+-----
+
+If you provide `--hooks <path>`, the module at `path` will be `required`. It is expected to export an object
+whose properties are hook functions. A hook function has the following signature:
+
+```javascript
+function(data, callback){ /* ... */ }
+```
+
+Here, `data` is a blob of data corresponding to the current state. Usually it's a set of useful metadata about
+the package currently being processed.
+
+The callback's signature is:
+
+```javascript
+function(error, shouldSave){ /* ... */ }
+```
+
+Where `shouldSave` is a boolean stating whether or not to actually perform the action that happens right
+after the hook (usually writing something to disk). In most cases, you'll want to call the callback with
+`callback(null, true)`. To prevent the action from happening, you can do `callback(null, false)`.
+
```

Comment:
We should add some docs here that the meta-data that we are passing into these hooks are by reference so if they modify that data that's what get's written.

Maybe we should move our domain rewrite code into the JSON hooks? The part that rewrites the domain name based on what their mirror name. This way the domain rewrite only happens if they don't override the hook. If they do, then it's their responsibility to handle the rewrite.